content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: 'Views'
+metaTitle: 'How to include views in your Prisma schema'
+metaDescription: 'How to include views in your Prisma schema'
+hidePage: false
+preview: true
+tocDepth: 2
+---
+
+<TopBlock>
+
+<Admonition type="warning">
+
+Support for views is currently a very early, experimental preview feature.
+
+</Admonition>
+
+<!-- TODO: add more to admonition -->
+
+Database views allow you to name and store a query. In relational databases, views are stored queries that may include columns in multiple tables. In MongoDB, views are queryable objects where the contents are defined by an [aggregation pipeline](https://www.mongodb.com/docs/manual/core/aggregation-pipeline) on other collections.
+
+This guide explains how to enable the `views` preview feature and add views to your Prisma schema.
+
+</TopBlock>
+
+## How to enable the <inlineCode>views</inlineCode> preview feature
+
+Support for views is currently in an early preview. To enable the `views` preview feature, add the `views` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema file:
+
+```prisma file=schema.prisma highlight=3;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["views"]
+}
+```
+
+## How to use views in your Prisma schema
+
+To add a view to your Prisma schema, use the `view` keyword:
+
+```prisma file=schema.prisma
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     Int    @id @default(autoincrement())
+  bio    String
+  user   User   @relation(fields: [userId], references: [id])
+  userId Int    @unique
+}
+
+// view UserInfo {
+// ... view syntax here...
+//  }
+```
```

Comment:
I think we should add the following general Prisma + views information:

- Need to write them into your Prisma schema by hand for now, `db pull` will not pick them up
- If you run `db pull` with an existing schema with `view`s, it will remove them and you need to make sure they are not removed by reverting that change
- Running Migrations via `db push` or `migrate *` commands will not create or run any SQL related to these views - you need to use `migrate dev --create-only` and add the SQL for them manually if you want Prisma to manage the views. 
- Alternatively you can create them manually outside of Prisma context
- Right now all views are assumed updateable and Prisma Client will give you all functionality it would give you for normal views

I think those are the important things people need to know so they are able to use the feature.

Reply:
True:
- Similar to `model`s Prisma also requires one field of a `view` to be marked as `@id`, `@unique` or multiple fields via `@@id` or `@@unique`. (We probably have a link we can use for that)

Reply:
Only thing not covered in current text is this now:
> Right now all views are assumed updateable and Prisma Client will give you all functionality it would give you for normal views

Reply:
How updates work is now covered here: https://github.com/prisma/docs/pull/4364/files#diff-168d613c226031f9a98a5560da26d2d080b2b22188985a02458d201944faa28aR180